Fix unit tests for AWS

| Affected integration |
|---|
| AWS |

# Description

While working on [this PR](https://github.com/wazuh/wazuh/pull/12630) we detected that some tests were not passing. The reason for the failure was that the unit test of the external integrations was trying to execute a binary that doesn't exist in the repository. The binary is `wazuh-control`, and the failure occurred when executing `test_config_format_created_date()` in `test_aws.py`

The output of the error is the following:

```
_____________ test_config_format_created_date[2021/1/19-20210119] ______________
utils.py:61: in call_wazuh_control
    proc = subprocess.Popen([wazuh_control, option], stdout=subprocess.PIPE)
/usr/local/lib/python3.9/subprocess.py:951: in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
/usr/local/lib/python3.9/subprocess.py:1823: in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
E   FileNotFoundError: [Errno 2] No such file or directory: '/tmp/build/wazuh/bin/wazuh-control'

During handling of the above exception, another exception occurred:
aws/tests/test_aws.py:325: in test_config_format_created_date
    bucket = aws_s3.AWSConfigBucket(**{'reparse': False, 'access_key': None, 'secret_key': None,
aws/aws_s3.py:1139: in __init__
    AWSLogsBucket.__init__(self, **kwargs)
aws/aws_s3.py:1043: in __init__
    AWSBucket.__init__(self, **kwargs)
aws/aws_s3.py:513: in __init__
    WazuhIntegration.__init__(self, access_key=access_key,
aws/aws_s3.py:150: in __init__
    self.wazuh_version = utils.get_wazuh_version()
utils.py:111: in get_wazuh_version
    return get_wazuh_info("WAZUH_VERSION")
utils.py:84: in get_wazuh_info
    wazuh_info = call_wazuh_control("info")
utils.py:66: in call_wazuh_control
    exit(1)
E   SystemExit: 1
----------------------------- Captured stdout call -----------------------------
ERROR: a problem occurred while executing /tmp/build/wazuh/bin/wazuh-control
```

## Tasks
- [x] Fix the test.
- [x] Test in a manager.
- [x] Test in an agent.
- [x] Unit tests without failures. Updated if there are any relevant changes.
- [x] API integration tests without failures. Updated if there are any relevant changes.
- [x] Update the documentation if necessary.
- [x] Add entry to the changelog if necessary.

--- a/wodles/aws/tests/conftest.py
+++ b/wodles/aws/tests/conftest.py
@@ -77,3 +77,20 @@ def aws_waf_bucket(request):
          patch('sqlite3.connect'), \
          patch('utils.get_wazuh_version'):
         return aws_s3.AWSWAFBucket(**{k: v for i in request.param for k, v in i.items()})
+
+
+@pytest.fixture(params=deepcopy(AWS_BUCKET_PARAMS))
+def aws_config_bucket(request):
+    """
+    Return a AWSConfigBucket instance.
+
+    Parameters
+    ----------
+    request : pytest.fixtures.SubRequest
+        Object that contains information about the current test.
+    """
+    with patch('aws_s3.AWSConfigBucket.get_client'), \
+         patch('aws_s3.AWSConfigBucket.get_sts_client'), \
+         patch('sqlite3.connect'), \
+         patch('utils.get_wazuh_version'):
+        return aws_s3.AWSConfigBucket(**{k: v for i in request.param for k, v in i.items()})

--- a/wodles/aws/tests/test_aws.py
+++ b/wodles/aws/tests/test_aws.py
@@ -302,13 +302,13 @@ def test_aws_waf_load_information_from_file_ko(
 
 
 @pytest.mark.parametrize('date, expected_date', [
-    ('2021/1/19','20210119'),
+    ('2021/1/19', '20210119'),
     ('2021/1/1', '20210101'),
     ('2021/01/01', '20210101'),
     ('2000/2/12', '20000212'),
     ('2022/02/1', '20220201')
 ])
-def test_config_format_created_date(date: str, expected_date: str):
+def test_config_format_created_date(date: str, expected_date: str, aws_config_bucket):
     """
     Test AWSConfigBucket's format_created_date method.
 
@@ -318,15 +318,7 @@ def test_config_format_created_date(date: str, expected_date: str):
         The date introduced.
     expected_date : str
         The date that the method should return.
+    aws_config_bucket : aws_s3.AWSConfigBucket
+        Instance of the AWSConfigBucket class.
     """
-    with patch('aws_s3.AWSConfigBucket.get_client'), \
-        patch('aws_s3.AWSConfigBucket.get_sts_client'), \
-        patch('sqlite3.connect'):
-        bucket = aws_s3.AWSConfigBucket(**{'reparse': False, 'access_key': None, 'secret_key': None,
-                        'profile': None, 'iam_role_arn': None, 'bucket': 'test',
-                        'only_logs_after': '19700101', 'skip_on_error': True,
-                        'account_alias': None, 'prefix': 'test',
-                        'delete_file': False, 'aws_organization_id': None,
-                        'region': None, 'suffix': '', 'discard_field': None,
-                        'discard_regex': None, 'sts_endpoint': None, 'service_endpoint': None})
-    assert bucket._format_created_date(date) == expected_date
+    assert aws_config_bucket._format_created_date(date) == expected_date

